types/mongodb/index.d.ts

```diff
@@ -103,54 +153,33 @@ export interface SocketOptions {
     socketTimeoutMS?: number;
 }
 
-//http://mongodb.github.io/node-mongodb-native/2.1/api/Server.html
-export interface ServerOptions {
-    // - specify the number of connections in the pool default:5
-    poolSize?: number;
-    ssl?: boolean;
-    sslValidate?: Object;
-    checkServerIdentity?: boolean | Function;
-    sslCA?: Array<Buffer | string>;
-    sslCert?: Buffer | string;
-    sslKey?: Buffer | string;
-    sslPass?: Buffer | string;
-    socketOptions?: SocketOptions;
+// http://mongodb.github.io/node-mongodb-native/2.2/api/Server.html
+export interface ServerOptions extends SSLOptions, HighAvailabilityOptions{
+    // Default: 30;
     reconnectTries?: number;
+    // Default: 1000;
     reconnectInterval?: number;
+    // Default: true;
+    monitoring?: boolean
+    socketOptions?: SocketOptions;
+}
+
+// http://mongodb.github.io/node-mongodb-native/2.2/api/Mongos.html
+export interface MongosOptions extends SSLOptions, HighAvailabilityOptions{
+    // Default: 15; Cutoff latency point in MS for MongoS proxy selection
+    acceptableLatencyMS?: number;
+    socketOptions?: SocketOptions;
 }
 
-//http://mongodb.github.io/node-mongodb-native/2.1/api/ReplSet.html
+// http://mongodb.github.io/node-mongodb-native/2.2/api/ReplSet.html
 export interface ReplSetOptions {
```

Comment:
I think `ReplSetOptions` should extend `SSLOptions` and `HighAvailabilityOptions`